### ISSUE TITLE
consider a new role=icon

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/aria#1088]
Icons are becoming more prevalent, particularly on mobile the sole way of labeling something. We are considering a new Success Criteria at WCAG 2.2 (just a proposal at this stage) called "Icon Descriptions" which could require a mechanism to expose the text alternative visually in these situations (i.e., a tooltip type of thing).

It might be a good idea to have role=icon in these situations were an icon is the label, rather than a generic img.

https://docs.google.com/document/d/1HzSsCGelWfz_Z-M7NyUzJOvl1A1kAStyl8epYdpZhoA/edit?usp=sharing

**Comment by scottaohara** [comment]
I could maybe see a synonym to `img` being setup for an icon role, but a new role itself seems unnecessary?

For instance, in situations like the test case linked in the google doc, an image/icon inside of a button element/role will get treated as `role=presentation`, so the usefulness of a role there is nill.

**Comment by jnurthen** [comment]
@DavidMacDonald what would you propose exposing in the Accessibility API which would be different to any other image?

**Comment by carmacleod** [comment]
@DavidMacDonald Are you saying that (in the example) `role="icon"` would go on the `button`?
If so, would the expectation be that screen readers could say something like "icon" or "info" instead of "button", so that screen reader users wouldn't activate the button? (I bet some screen readers would still say "clickable", though).

I've seen different interaction patterns with these things. Some need to be clicked/tapped to show their "tooltip", and others show it on focus/hover/(maybe long tap?).

You mentioned that these have been used as "the sole way of labeling something", but the example doesn't show the "something" being labelled. Do you mean that the `i` icon would be placed after a text input or similar?

Just trying to understand what is being asked. :)

**Comment by AmeliaBR** [comment]
Related: the [`graphics-symbol` role](https://w3c.github.io/graphics-aria/#graphics-symbol) from the initial ARIA Graphics module.

**Comment by DavidMacDonald** [comment]
@jnurthen @carmacleod 
I'm proposing that instead of being announced as an image, it would be announced as an icon, which is of course a very particular king of image.
Perhaps we could manage this with an attribute that specifies the type of image like we do with 
```role=dialog aria-modal="true" ```
Some idea options
```
<img aria-icon="true" alt=......>
<i role=icon class="fa-warning" aria-label="warning">...></i>
<img ...role="icon" alt="warning">
<button ...><img ...role="icon" alt="edit address"></button>
<img aria-imagetype="icon">
<img aria-imagetype="photo">
<img aria-imagetype="chart">
<span role="image" aria-imagetype="icon" class="cssimage">...
<span role="image" aria-imagetype="photo" >
<span role="image" aria-imagetype="chart">
```
The API could have a new bucket for image type.

**Comment by jnurthen** [comment]
I don't think this fits into any of the projects we are currently looking at for 1.3
Adding a 1.4 Milestone on this so it can be evaluated in the future.

**Comment by DavidMacDonald** [comment]
@jnurthen 
What about proposal 2 above?
aria-imagetype="[token]"

**Comment by jnurthen** [comment]
I don't think it really fits into any of the work we have planned for 1.3
If someone wants to pick it up and push it through then they are welcome to do so. Do you have any indication from AT developers that they want a way to support something like this. If we get that and commitments from AT to support such a feature then we would prioritise it higher but without this I don't want to spend time on it.

**Comment by scottaohara** [comment]
proposal 2 wouldn't help with the icon still not being exposed as an icon within a button though?

**Comment by jnurthen** [comment]
Would 
```
role="image" aria-roledescripton="icon"
```
not work for this use case?
